sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -193,38 +195,68 @@ case class InMemoryTableScanExec(
 
   private val inMemoryPartitionPruningEnabled = sqlContext.conf.inMemoryPartitionPruning
 
+  private def doFilterCachedBatches(
+      rdd: RDD[CachedBatch],
+      partitionStatsSchema: Seq[AttributeReference]): RDD[CachedBatch] = {
+    val schemaIndex = partitionStatsSchema.zipWithIndex
+    rdd.mapPartitionsWithIndex {
+      case (partitionIndex, cachedBatches) =>
+        if (inMemoryPartitionPruningEnabled) {
+          cachedBatches.filter { cachedBatch =>
+            val partitionFilter = newPredicate(
+              partitionFilters.reduceOption(And).getOrElse(Literal(true)),
+              partitionStatsSchema)
+            partitionFilter.initialize(partitionIndex)
+            if (!partitionFilter.eval(cachedBatch.stats)) {
```

Comment:
Are there any issues with discarding a partition based on statistics that could be partially computed (e.g. when total size in bytes of a partition iterator is larger than configurable batch size) as per https://github.com/apache/spark/pull/19810/files#diff-5fc188468d3066580ea9a766114b8f1dR74? 

Would be it be beneficial to record such situation by logging it, and still include such partition when statistics are partially computed and filters are evaluated to false, or discard all statistics when some of the partitions hit this situation? Thanks!

Reply:
I might not understand your proposal well...are you trying to simplify the logic in https://github.com/apache/spark/pull/19810/files#diff-5fc188468d3066580ea9a766114b8f1dR74? it would make the code simpler but degrade pruning effect here,

Reply:
All good, no need to change. I was trying to understand the code, so my question would be referring to statistics collection overall, not changes in this PR. Link points to a condition (exists before this PR) that could potentially result in exiting iterator before exhausting all records in it, so statistics would be partially collected, which might affect any filtering that uses such statistics - though it is quite possibly handled later, or a theoretical use case.

Reply:
@sadikovi this while loop is building CatchedBatch, it just decides what's the right time to seal the building window of a CatchedBatch and start the next one....so, in any way, you need to go through every all records in the partition,